src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -289,6 +289,8 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                 // Core WebJobs/Script Host services
                 services.AddSingleton<ScriptHost>();
 
+                services.AddSingleton<IHttpProxyService, DefaultHttpProxyService>();
```

Comment:
Could this be conditionally registered? Only when custom handlers is enabled?

Reply:
With the cleanup and refactor, this service will also be used for HTTP Proxying for worker http streaming. Information about streaming support/capabilities won't be known until there is a worker init response (which occurs much later than this stage).